### ISSUE TITLE
Unfortunately doesn't work on facebook news thumbnails.

### ISSUE THREAD

**Issue by peeter-t2** [tomroyal/Make-America-Kittens-Again#13]
Any chance this could be prevented in some future update?

If this is difficult, perhaps easier would be to automatically insert kitten pictures when posting a news article for anyone who has this app turned on?

Thanks for the life-saving application!

**Comment by tomroyal** [comment]
I'm afraid Facebook is about the toughest site I've tried to kittenify so far - I think MAKA currently has about a 10% success rate there. Will re-open this and let you know if I can get a higher hit-rate :)

**Comment by peeter-t2** [comment]
Naturally, some fortresses will hold out longer... ;) Thanks for all the great work! I will definitely look forward to any updates! :)

**Comment by peeter-t2** [comment]
I was really happy when the app worked once in fb, and then had a look at the source how it works. My java is really rusty, and I haven't done much with web, but something like this could work:

Given that fb link images are embedded within links, I counted that it's the third child in, but this could be adapted.

Find images (as you do);
If third-parent is an "a" - add also contents of the link url for checking.
If link url contains blist, change that image too.

I'm not sure if this could become unreasonably greedy in other contexts and if it would work, potentially, just a website check could be added. I added a fork with an idea where to put it, but that version probably won't work, as I don't really know how to test the script.

Good luck!
